### DISCUSSION TITLE
[V5] how to get values from ManyToMany relationship?

### DISCUSSION THREAD

**Discussion by furatamasensei** [adonisjs/core#1768] [Help] [answered]
I'm trying to make a seeder to test my API. I want to create a Company along with its Branch, and I've created something like this in for the seeder:

```ts
// somewhere inside Role.ts
@manyToMany(() => User)
public users: ManyToMany<typeof User>

// CompanySeeder.ts
export default class CompanySeeder extends BaseSeeder {
  public static developmentOnly = true

  public async run () {
    const rcl = await Role.findByOrFail('name', 'c-level')
    for (const user of rcl.users) {
      const companies = await CompanyFactory.merge({ userId: user.id }).createMany(5)

      for (const company of companies) {
        await BranchFactory.merge({ companyId: company.id }).createMany(5)
      }
    }
  }
}
```

But the console give me an error: `rcl.users is not iterable`. I tried to preload but it can't be used since the type is `void`. Is there an alternative way to do this?

**Comment by melokki** [comment, marked as answer]
the preload method does not return the users array, it just preloads the data.
so, in your example you should be able to do 
```ts
await rcl.preload('users');
```
and after that, on your next line you should have access to your users array to iterate over it

**Reply by furatamasensei** [reply]
Thank you for replying.. It works now, thanks!

**Comment by mcordoba** [comment]
You can also do:

```javascript
const users = await rcl.related('users').query().fetch()
```
